spec.html

```diff
@@ -32422,38 +32179,40 @@ <h1>AtomEscape</h1>
+          1. Return ! CharacterSetMatcher(_A_, *false*, _direction_).
         </emu-alg>
         <emu-note>
           <p>An escape sequence of the form `\\` followed by a nonzero decimal number _n_ matches the result of the _n_<sup>th</sup> set of capturing parentheses (<emu-xref href="#sec-notation"></emu-xref>). It is an error if the regular expression has fewer than _n_ capturing parentheses. If the regular expression has _n_ or more capturing parentheses but the _n_<sup>th</sup> one is *undefined* because it has not captured anything, then the backreference always succeeds.</p>
         </emu-note>
         <p>The production <emu-grammar>AtomEscape :: `k` GroupName</emu-grammar> evaluates as follows:</p>
         <emu-alg>
-          1. Search the enclosing |Pattern| for an instance of a |GroupSpecifier| for a |RegExpIdentifierName| which has a StringValue equal to the StringValue of the |RegExpIdentifierName| contained in |GroupName|.
+          1. Search the enclosing |Pattern| for an instance of a |GroupSpecifier| containing a |RegExpIdentifierName| which has a StringValue equal to the StringValue of the |RegExpIdentifierName| contained in |GroupName|.
           1. Assert: A unique such |GroupSpecifier| is found.
-          1. Let _parenIndex_ be the number of left-capturing parentheses in the entire regular expression that occur to the left of the located |GroupSpecifier|. This is the total number of <emu-grammar>Atom :: `(` GroupSpecifier Disjunction `)`</emu-grammar> Parse Nodes prior to or enclosing the located |GroupSpecifier|.
-          1. Call BackreferenceMatcher(_parenIndex_, _direction_) and return its Matcher result.
+          1. Let _parenIndex_ be the number of left-capturing parentheses in the entire regular expression that occur to the left of the located |GroupSpecifier|. This is the total number of <emu-grammar>Atom :: `(` GroupSpecifier Disjunction `)`</emu-grammar> Parse Nodes prior to or enclosing the located |GroupSpecifier|, including its immediately enclosing |Atom|.
+          1. Return ! BackreferenceMatcher(_parenIndex_, _direction_).
         </emu-alg>
 
         <emu-clause id="sec-backreference-matcher" aoid="BackreferenceMatcher">
           <h1>Runtime Semantics: BackreferenceMatcher ( _n_, _direction_ )</h1>
           <p>The abstract operation BackreferenceMatcher takes arguments _n_ (an integer) and _direction_ (an integer). It performs the following steps when called:</p>
           <emu-alg>
+            1. Assert: _n_ &gt; 0.
             1. Return a new Matcher with parameters (_x_, _c_) that captures _n_ and _direction_ and performs the following steps when called:
               1. Assert: _x_ is a State.
               1. Assert: _c_ is a Continuation.
               1. Let _cap_ be _x_'s _captures_ List.
+              1. NOTE: _cap_ is indexed from 1 (<emu-xref href="#sec-notation"></emu-xref>).
               1. Let _s_ be _cap_[_n_].
```

Comment:
As long as we're here, I'd be happier if we added an explicit bound check, as in

```
1. If the number of elements in _cap_ is less than _n_, let _s_ be *undefined*.
1. Let _s_ be _cap_[_n_].
```

Reply:
The length of _cap_ is always _NcapturingParens_ by both definition and assignment, and there's a fair amount of other code that reads directly from it. I'd like to leave this unless you insist upon adding an assertion.

Reply:
Ah, that's fine then. Somehow I convinced myself that wasn't necessarily so.